3rdparty/xinputcheck-src/xinputcheck.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2013 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+
+#include "xinputcheck.h"
+
+#define SDL_bool      bool
+#define SDL_TRUE      true
+#define SDL_FALSE     false
+
+#define SDL_free         free
+#define SDL_malloc       malloc
+#define SDL_OutOfMemory  abort
+#define SDL_strstr       strstr
+#define SDL_memcmp       memcmp
+
+#define SDL_arraysize(array)    (sizeof(array)/sizeof(array[0]))
+
+static PRAWINPUTDEVICELIST SDL_RawDevList = NULL;
```

Comment:
Not that fond of that static caching here. It's a bit annoying for leak checks to begin with but I think what we will hit sooner or later is people hotplugging their gamepads making this break because it's relying on cached data from the first enumeration. We should probably pass through ```uiNewHardware``` changes to the enumeration machinery so it can reset the cache (haven't checked if ```uiNewHardware``` is actually robust).